src/core_init_atmosphere/mpas_parse_geoindex.F

```diff
@@ -0,0 +1,265 @@
+module mpas_parse_geoindex
+
+   use mpas_log, only : mpas_log_write
+   use mpas_derived_types, only : MPAS_LOG_ERR, MPAS_LOG_WARN
+   use mpas_pool_routines
+
+   implicit none
+
+   private
+
+   public :: mpas_parse_index
+
+   contains
+
+   !***********************************************************************
+   !
+   !  subroutine mpas_parse_geoindex
+   !
+   !> \brief   Parse a geogrid's index file and put the results into an MPAS pool
+   !> \author  Miles A. Curry
+   !> \date    02/20/2020
+   !> \details
+   !> Parse an index file of a static data set into an MPAS pool, allocating
+   !> each keyword=value pair into the pool with the pool member key being
+   !> keyword, and the value being value.
+   !>
+   !> This function can parse index files with one keyword=value pair
+   !> per line; a "#" at the start of a line, which will cause the line to be
+   !> ignored; or an empty line containing only a newline/return character, which
+   !> will also be ignored. Spaces or tabs before, between or after the
+   !> keyword=value tokens are > ignored.
+   !>
+   !> If a line contains anything but the above valid syntaxes, a syntax
+   !> error will raised and -1 will be returned.
+   !>
+   !> Case is ignored.
+   !>
+   !> The definitions of a keyword, which can found in section 3-53
+   !> of the WRF-AWR User's Guide, will determine the corresponding type
+   !> of that keyword. A keyword that has been assigned the wrong type
+   !> will raise a type error and -1 will be returned.
+   !>
+   !> Keywords that are not handled explicitly by this function will produce
+   !> a warning that the keyword is unrecognized.
+   !
+   !-----------------------------------------------------------------------
+   function mpas_parse_index(path, geo_pool) result(ierr)
+
+      use mpas_io_units
+
+      implicit none
+      ! Input Variables
+      character (len=StrKIND), intent(in) :: path
+      type (mpas_pool_type), intent(inout), pointer :: geo_pool
```

Comment:
Ultimately it probably doesn't matter, but I don't think we actually need to require that `geo_pool` be a pointer. This would in principle allow calling code to pass an actual argument that was not a pointer; and, it avoids the need verify that the `geo_pool` pointer is associated in this routine.

Reply:
I think that's a good idea. There isn't any reason for it to be a pointer. I'll remove the `pointer` attribute from `geo_pool`.